frontend/src/Game/Match.tsx

```diff
@@ -42,35 +73,83 @@ function Ball(props: { pBall: ball }): ReactElement {
   )
 }
 
-function updateBall(pBall: ball, deltaTime: number, speed: number): ball {
+function updateBall(
+  pBall: IBall,
+  deltaTime: number,
+  speed: number,
+  leftPaddle: IPaddle,
+  rightPaddle: IPaddle,
+  handleScoreChange: () => void
+): IBall {
   pBall.pos.x += pBall.vel.x * deltaTime * speed
   pBall.pos.y += pBall.vel.y * deltaTime * speed
   if (pBall.pos.y <= 0 && pBall.vel.y < 0) {
     pBall.vel.y *= -1
-  }
-  if (pBall.pos.y >= hght - ballPx && pBall.vel.y > 0) {
+  } else if (pBall.pos.y >= hght - ballPx && pBall.vel.y > 0) {
     pBall.vel.y *= -1
-  }
-  if (pBall.pos.x <= 0 && pBall.vel.x < 0) {
+  } else if (
+    // goal hit
+    (pBall.pos.x <= 0 && pBall.vel.x < 0) ||
+    (pBall.pos.x >= wid - ballPx && pBall.vel.x > 0)
+  ) {
+    pBall.pos = deepCpInitBall().pos
     pBall.vel.x *= -1
-  }
-  if (pBall.pos.x >= wid - ballPx && pBall.vel.x > 0) {
+    handleScoreChange()
+  } else if (
+    // paddle hit
+    (leftPaddle.pos.x <= pBall.pos.x + ballPx &&
```

Comment:
paddle hitはvelocityもで決める必要があると思います
これがバグの原因かもしれません？ボールが跳ね続けるのは、どうしたって方向を変えてしまうからかな。
しかし、ボールがパドルに向かっているときだけ、方向を変える必要があると思うのです。
すでにパドルから離れている場合は、方向を変える必要はないとおもいます。
```suggestion
    (pBall.vel.x < 0 && .......)
```